content/developers/developer-patterns/navigating-merklelogs/index.md

```diff
@@ -393,42 +320,25 @@ Using our "canonical" mmr log for illustration, we get this diagram
 
-## The "look back" peak stack can be visualised like this
+## The "look back" peak stack can be visualized like this
 
-A specific challenge for log implementations with very large data sets is then
-answering "how far back" or "how far forward" may I need to look ?
+A specific challenge for log implementations with very large data sets is  answering "how far back" or "how far forward" may I need to look?
 
-MMR's differ from classic binary merkle trees in how the incomplete sub trees
-are combined into a common root. For an MMR, the common root is defined by an
-algorithm for combining the, adjacent, incomplete sub trees. Rather than by the
-more traditional, temporary, assignment of un-balanced siblings. Such
-un-balanced siblings would later have to be re-assigned (balanced) when there
+MMRs differ from classic binary Merkle trees in how the incomplete sub trees are combined into a common root. For an MMR, the common root is defined by an algorithm for combining the, adjacent, incomplete sub trees. Rather than by the more traditional, temporary, assignment of un-balanced siblings. Such un-balanced siblings would later have to be re-assigned (balanced) when there
 were sufficient leaves to merge the sub trees.
 
-This detail is what permits us to publish the log data immediately that your
-events are added.
+This detail is what permits us to publish the log data immediately that your events are added.
 
-So the specific properties of Merkle Mountain Ranges lead to an efficiently
-computable and stable answer to the questions of "which other nodes do I need".
-Such that we *know* categorically we do not need to look *forward* of the
-current massif. And further we know precisely which nodes we need from the
-previous massifs.
+So the specific properties of Merkle Mountain Ranges lead to an efficiently computable and stable answer to the questions of "which other nodes do I need". Such that we *know* categorically we do not need to look *forward* of the current massif and further we know precisely which nodes we need from the previous massifs.
 
-The "free nodes" in the alpine zone always require "ancestors" from previous
-nodes when producing inclusion proofs that pass through them, and when adding
-new nodes to the end of the log. Here we can see they are very predictable (can
-be calculated without reference to the tree data). We accumulate these peaks in
-a stack because the pop order is the order we need them when adding leaves at
-the end of the log.
+The "free nodes" in the alpine zone always require "ancestors" from previous nodes when producing inclusion proofs that pass through them, and when adding new nodes to the end of the log. Here we can see they are very predictable (can be calculated without reference to the tree data). We accumulate these peaks in a stack because the pop order is the order we need them when adding leaves at the end of the log.
```

Comment:
```suggestion
The "free nodes" in the alpine zone always require "ancestors" from previous nodes when producing inclusion proofs that pass through them, and when adding new nodes to the end of the log.
Here we can see they are very predictable (can be calculated without reference to the tree data).
We accumulate these peaks in a stack because the pop order is the order we need them when adding leaves at the end of the log.
```